source/code/plugins/in_npmd.rb

```diff
@@ -0,0 +1,676 @@
+                    else
+                        _ipConfig.add_attribute("Mask", maskHash[ip["SubnetName"]])
+                        _agent.elements << _ipConfig
+                    end
+                end
+                _agents.elements << _agent
+            end
+            _agents
+        end
+
+        def self.createNetworkElements(networkArray, subnetIdHash)
+            _networks = REXML::Element.new("Networks")
+            networkArray.each do |x|
+                _network = REXML::Element.new("Network")
+                _network.add_attribute("Name", x["Name"])
+                x["Subnets"].each do |sn|
+                    _subnetId = subnetIdHash[sn]
+                    if _subnetId.nil?
+                        Logger::logWarn "Did not find subnet id for subnet name #{sn} in hash", 2*Logger::loop
+                    else
+                        _snConfig = REXML::Element.new("Subnet")
+                        _snConfig.add_attribute("ID", subnetIdHash[sn])
+                        _snConfig.add_attribute("Disabled", "False")
+                        _snConfig.add_attribute("Tag", "")
+                        _network.elements << _snConfig
+                    end
+                end
+                _networks.elements << _network
+            end
+            _networks
+        end
+
+        def self.createRuleElements(ruleArray)
+            _rules = REXML::Element.new("Rules")
+            ruleArray.each do |x|
+                _rule = REXML::Element.new("Rule")
+                _rule.add_attribute("Name", x["Name"])
+                _rule.add_attribute("Description", "")
+                _rule.add_attribute("Protocol", x["Protocol"])
+                _alertConfig = REXML::Element.new("AlertConfiguration")
+                _alertConfig.add_element("Loss", {"Threshold" => x["LossThreshold"] })
+                _alertConfig.add_element("Latency", {"Threshold" => x["LatencyThreshold"]})
+                _netTestMtx = REXML::Element.new("NetworkTestMatrix")
+                x["Rules"].each do |r|
+                    _snPair = REXML::Element.new("SubnetPair")
+                    _snPair.add_attribute("SourceSubnet", r["SS"])
+                    _snPair.add_attribute("SourceNetwork", r["SN"])
+                    _snPair.add_attribute("DestSubnet", r["DS"])
+                    _snPair.add_attribute("DestNetwork", r["DN"])
+                    _netTestMtx.elements << _snPair
+                end
+                _rule.elements << _alertConfig
+                _rule.elements << _netTestMtx
+                _rules.elements << _rule
+            end
+            _rules
+        end
+    end
+
+    # This class holds the methods for parsing
+    # a config sent via DSC into a hash
+    class UIConfigParser
+        public
+
+        # Only accessible method
+        def self.parse(string)
+            begin
+                _doc = REXML::Document.new(string)
```

Comment:
we suspect the REXML class has some memory issue. can you check your memory usage?